### ISSUE TITLE
no 2.1.0-rc1 or 2.1.0-rc2 in dockerhub only 2.0.0

### ISSUE THREAD

**Issue by johngrabner** [tensorflow/tensorflow#35364]
**System information**
- TensorFlow version: 2.1.0

**Describe the problem**
Tensorflow/tensorflow does not list 2.1.0-rc1 or rc2 in the list of available tags.
These are available on colab but somehow did not make it to dockerhub.

**Provide the exact sequence of commands / steps that you executed before running into the problem**
https://hub.docker.com/r/tensorflow/tensorflow/tags/?page=1&name=2.1.0
observe no tags

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by johngrabner** [comment]
I tried to build the docker from git but get the following error
```
docker build --build-arg USE_PYTHON_3_NOT_2=true --build-arg TF_PACKAGE_VERSION=2.1.0-rc2  -f ./dockerfiles/gpu.Dockerfile -t tf .
docker run --gpus all -it tf
python3
import tensorflow as tf
2019-12-26 14:45:30.126243: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.6
2019-12-26 14:45:30.126620: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libnvinfer_plugin.so.6'; dlerror: libnvrtc.so.10.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/cuda/extras/CUPTI/lib64
2019-12-26 14:45:30.126633: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:30] Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
```
Trying various things to fix the above, but no success so far.

**Comment by angerson** [comment]
@johngrabner Thanks for your report. #35367 should have fixed this -- it seems like the images were never updated to support CUDA 10.1, which TensorFlow 2.1 requires. Can you try `git pull`?

I'm waiting on some updates to our internal CI to be able to push the new images. Hopefully, they'll be available before the end of the day.

**Comment by johngrabner** [comment]
Looks like image got uploaded
```
2.1.0rc2-gpu
docker pull tensorflow/tensorflow:2.1.0rc2-gpu
Last updated10 minutes agobytensorflowpackages
```

Trying it out now

**Comment by johngrabner** [comment]
I think it is working, but there is a warning on the second line

2019-12-26 21:53:03.304048: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.6
2019-12-26 21:53:03.304418: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libnvinfer_plugin.so.6'; dlerror: libnvrtc.so.10.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
2019-12-26 21:53:03.304431: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:30] Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2019-12-26 21:53:08.633799: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2019-12-26 21:53:08.656459: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.656893: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.755GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2019-12-26 21:53:08.656950: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 21:53:08.657003: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 21:53:08.658244: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2019-12-26 21:53:08.658569: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2019-12-26 21:53:08.659846: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2019-12-26 21:53:08.660518: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2019-12-26 21:53:08.660557: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 21:53:08.660628: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.661080: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.661436: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2019-12-26 21:53:08.661693: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-12-26 21:53:08.682339: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3696000000 Hz
2019-12-26 21:53:08.682953: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x87addb0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2019-12-26 21:53:08.682968: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2019-12-26 21:53:08.760651: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.761034: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x8159cc0 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2019-12-26 21:53:08.761048: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): GeForce RTX 2080 Ti, Compute Capability 7.5
2019-12-26 21:53:08.761157: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.761519: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.755GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2019-12-26 21:53:08.761548: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 21:53:08.761559: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 21:53:08.761569: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2019-12-26 21:53:08.761579: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2019-12-26 21:53:08.761593: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2019-12-26 21:53:08.761607: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2019-12-26 21:53:08.761642: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 21:53:08.761696: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.762104: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.762428: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2019-12-26 21:53:08.762457: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 21:53:08.931909: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1096] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-12-26 21:53:08.931936: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102]      0 
2019-12-26 21:53:08.931942: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] 0:   N 
2019-12-26 21:53:08.932156: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.932582: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 21:53:08.932950: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1241] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8288 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5)

**Comment by johngrabner** [comment]
It there a standard test you normally run.

The sample code I am using is not happy with rc2

sudo docker exec -it 282c2ff49852 /bin/bash

________                               _______________                
___  __/__________________________________  ____/__  /________      __
__  /  _  _ \_  __ \_  ___/  __ \_  ___/_  /_   __  /_  __ \_ | /| / /
_  /   /  __/  / / /(__  )/ /_/ /  /   _  __/   _  / / /_/ /_ |/ |/ / 
/_/    \___//_/ /_//____/ \____//_/    /_/      /_/  \____/____/|__/


You are running this container as user with ID 1000 and group 1000,
which should map to the ID and group for your user on the Docker host. Great!

tf-docker /tf > python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
2019-12-26 22:10:33.702129: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.6
2019-12-26 22:10:33.703898: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] **Could not load dynamic library 'libnvinfer_plugin.so.6'; dlerror: libnvrtc.so.10.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
2019-12-26 22:10:33.703956: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:30] Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.**
>>> print(tf.__version__)
2.1.0-rc2
>>> device_lib.list_local_devices()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'device_lib' is not defined
>>> sess = tf.Session(config=tf.ConfigProto(log_device_placement=True))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'tensorflow' has no attribute 'Session'
>>> from tensorflow.python.client import device_lib
>>> print(device_lib.list_local_devices())
2019-12-26 22:14:23.959816: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-12-26 22:14:24.005268: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3696000000 Hz
2019-12-26 22:14:24.006412: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x596fee0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2019-12-26 22:14:24.006426: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2019-12-26 22:14:24.008570: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2019-12-26 22:14:24.122916: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:14:24.123297: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x59a2520 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2019-12-26 22:14:24.123312: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): GeForce RTX 2080 Ti, Compute Capability 7.5
2019-12-26 22:14:24.123431: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:14:24.123737: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.755GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2019-12-26 22:14:24.123767: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 22:14:24.123791: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 22:14:24.141434: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2019-12-26 22:14:24.144722: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2019-12-26 22:14:24.189225: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2019-12-26 22:14:24.196701: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2019-12-26 22:14:24.196859: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 22:14:24.197215: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:14:24.199081: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:14:24.200805: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2019-12-26 22:14:24.201544: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 22:14:24.827415: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1096] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-12-26 22:14:24.827445: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102]      0 
2019-12-26 22:14:24.827451: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] 0:   N 
2019-12-26 22:14:24.827988: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:14:24.828372: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:14:24.828708: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1241] Created TensorFlow device (/device:GPU:0 with 9441 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5)
[name: "/device:CPU:0"
device_type: "CPU"
memory_limit: 268435456
locality {
}
incarnation: 12645776964756662951
, name: "/device:XLA_CPU:0"
device_type: "XLA_CPU"
memory_limit: 17179869184
locality {
}
incarnation: 8039547365867512922
physical_device_desc: "device: XLA_CPU device"
, name: "/device:XLA_GPU:0"
device_type: "XLA_GPU"
memory_limit: 17179869184
locality {
}
incarnation: 14991733659365383547
physical_device_desc: "device: XLA_GPU device"
, name: "/device:GPU:0"
device_type: "GPU"
memory_limit: 9900208948
locality {
  bus_id: 1
  links {
  }
}
incarnation: 15540725193523866116
physical_device_desc: "device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5"
]
>>> exit
Use exit() or Ctrl-D (i.e. EOF) to exit
>>> exit()
tf-docker /tf > 
tf-docker /tf > cd /home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src/
tf-docker /home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src > python main.py --source=bentham --train
2019-12-26 22:15:29.610690: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.6
2019-12-26 22:15:29.610942: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libnvinfer_plugin.so.6'; dlerror: libnvrtc.so.10.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
2019-12-26 22:15:29.610972: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:30] Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2019-12-26 22:15:35.159777: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2019-12-26 22:15:35.197746: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.198250: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.755GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2019-12-26 22:15:35.198296: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 22:15:35.198384: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 22:15:35.199606: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2019-12-26 22:15:35.199860: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2019-12-26 22:15:35.201211: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2019-12-26 22:15:35.201923: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2019-12-26 22:15:35.201963: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 22:15:35.202050: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.202537: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.203006: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2019-12-26 22:15:35.203418: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-12-26 22:15:35.229352: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3696000000 Hz
2019-12-26 22:15:35.229900: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x894c1f0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2019-12-26 22:15:35.229913: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2019-12-26 22:15:35.305420: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.305824: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x88bd8d0 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2019-12-26 22:15:35.305839: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): GeForce RTX 2080 Ti, Compute Capability 7.5
2019-12-26 22:15:35.306028: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.306367: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.755GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2019-12-26 22:15:35.306395: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 22:15:35.306405: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 22:15:35.306417: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2019-12-26 22:15:35.306430: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2019-12-26 22:15:35.306447: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2019-12-26 22:15:35.306484: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2019-12-26 22:15:35.306507: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 22:15:35.306572: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.306891: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.307176: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2019-12-26 22:15:35.307206: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 22:15:35.477274: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1096] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-12-26 22:15:35.477303: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102]      0 
2019-12-26 22:15:35.477309: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] 0:   N 
2019-12-26 22:15:35.477490: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.477903: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 22:15:35.478271: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1241] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 9440 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5)
Model: "model"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input (InputLayer)           [(None, 1024, 128, 1)]    0         
_________________________________________________________________
conv2d (Conv2D)              (None, 512, 64, 16)       160       
_________________________________________________________________
p_re_lu (PReLU)              (None, 512, 64, 16)       16        
_________________________________________________________________
batch_normalization (BatchNo (None, 512, 64, 16)       112       
_________________________________________________________________
full_gated_conv2d (FullGated (None, 512, 64, 16)       4640      
_________________________________________________________________
conv2d_1 (Conv2D)            (None, 512, 64, 32)       4640      
_________________________________________________________________
p_re_lu_1 (PReLU)            (None, 512, 64, 32)       32        
_________________________________________________________________
batch_normalization_1 (Batch (None, 512, 64, 32)       224       
_________________________________________________________________
full_gated_conv2d_1 (FullGat (None, 512, 64, 32)       18496     
_________________________________________________________________
conv2d_2 (Conv2D)            (None, 256, 16, 40)       10280     
_________________________________________________________________
p_re_lu_2 (PReLU)            (None, 256, 16, 40)       40        
_________________________________________________________________
batch_normalization_2 (Batch (None, 256, 16, 40)       280       
_________________________________________________________________
full_gated_conv2d_2 (FullGat (None, 256, 16, 40)       28880     
_________________________________________________________________
dropout (Dropout)            (None, 256, 16, 40)       0         
_________________________________________________________________
conv2d_3 (Conv2D)            (None, 256, 16, 48)       17328     
_________________________________________________________________
p_re_lu_3 (PReLU)            (None, 256, 16, 48)       48        
_________________________________________________________________
batch_normalization_3 (Batch (None, 256, 16, 48)       336       
_________________________________________________________________
full_gated_conv2d_3 (FullGat (None, 256, 16, 48)       41568     
_________________________________________________________________
dropout_1 (Dropout)          (None, 256, 16, 48)       0         
_________________________________________________________________
conv2d_4 (Conv2D)            (None, 128, 4, 56)        21560     
_________________________________________________________________
p_re_lu_4 (PReLU)            (None, 128, 4, 56)        56        
_________________________________________________________________
batch_normalization_4 (Batch (None, 128, 4, 56)        392       
_________________________________________________________________
full_gated_conv2d_4 (FullGat (None, 128, 4, 56)        56560     
_________________________________________________________________
dropout_2 (Dropout)          (None, 128, 4, 56)        0         
_________________________________________________________________
conv2d_5 (Conv2D)            (None, 128, 4, 64)        32320     
_________________________________________________________________
p_re_lu_5 (PReLU)            (None, 128, 4, 64)        64        
_________________________________________________________________
batch_normalization_5 (Batch (None, 128, 4, 64)        448       
_________________________________________________________________
max_pooling2d (MaxPooling2D) (None, 128, 2, 64)        0         
_________________________________________________________________
reshape (Reshape)            (None, 128, 128)          0         
_________________________________________________________________
bidirectional (Bidirectional (None, 128, 256)          198144    
_________________________________________________________________
time_distributed (TimeDistri (None, 128, 128)          32896     
_________________________________________________________________
bidirectional_1 (Bidirection (None, 128, 256)          198144    
_________________________________________________________________
time_distributed_1 (TimeDist (None, 128, 98)           25186     
=================================================================
Total params: 692,850
Trainable params: 691,570
Non-trainable params: 1,280
_________________________________________________________________
Train for 545 steps, validate for 85 steps
Epoch 1/1000
2019-12-26 22:15:43.691246: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 22:15:44.122500: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 22:15:45.384099: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2019-12-26 22:15:45.395668: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2019-12-26 22:15:45.396141: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Unknown: Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[{{node model/conv2d/Conv2D}}]]
	 [[loss/time_distributed_1_loss/Log/_98]]
2019-12-26 22:15:45.396203: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Unknown: Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[{{node model/conv2d/Conv2D}}]]
  1/545 [..............................] - ETA: 1:01:25WARNING:tensorflow:Can save best model only with val_loss available, skipping.
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: 
WARNING:tensorflow:Reduce LR on plateau conditioned on metric `val_loss` which is not available. Available metrics are: lr
  1/545 [..............................] - ETA: 1:02:00Traceback (most recent call last):
  File "main.py", line 155, in <module>
    verbose=1)
  File "/home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src/network/model.py", line 175, in fit
    use_multiprocessing=use_multiprocessing, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training.py", line 819, in fit
    use_multiprocessing=use_multiprocessing)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 342, in fit
    total_epochs=epochs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 128, in run_one_epoch
    batch_outs = execution_function(iterator)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 98, in execution_function
    distributed_function(input_fn))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 568, in __call__
    result = self._call(*args, **kwds)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 632, in _call
    return self._stateless_fn(*args, **kwds)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 2363, in __call__
    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 1611, in _filtered_call
    self.captured_inputs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 1692, in _call_flat
    ctx, args, cancellation_manager=cancellation_manager))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 545, in call
    ctx=ctx)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/execute.py", line 67, in quick_execute
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.UnknownError: 2 root error(s) found.
  (0) Unknown:  Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[node model/conv2d/Conv2D (defined at /home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src/network/model.py:175) ]]
	 [[loss/time_distributed_1_loss/Log/_98]]
  (1) Unknown:  Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[node model/conv2d/Conv2D (defined at /home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src/network/model.py:175) ]]
0 successful operations.
0 derived errors ignored. [Op:__inference_distributed_function_17754]

Function call stack:
distributed_function -> distributed_function

2019-12-26 22:15:45.548644: W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Cancelled: Operation was cancelled

**Comment by johngrabner** [comment]
is there a way for me to pull down #35434 and see if it fixes the issue?

**Comment by johngrabner** [comment]
I don't think that fix is sufficient.

I added the same lines you added but to my dockerfile
```
FROM tensorflow/tensorflow:2.1.0rc2-gpu-py3-jupyter

# to see if  #35434 fixes the issues
ARG CUDA=10.1
RUN apt-get update && apt-get install -y --no-install-recommends \
        cuda-nvrtc-${CUDA/./-} \
        cuda-nvrtc-dev-${CUDA/./-} 

# stuff i need for the app
RUN pip install autopep8==1.4.4
RUN pip install editdistance==0.5.3
RUN pip install flake8==3.7.9
RUN pip install kaldiio==2.15.0
RUN pip install numba==0.46.0
RUN ["apt-get", "install", "-y", "libsm6", "libxext6", "libxrender-dev"]
RUN pip install opencv-python==4.1.2.30
```

to make sure docker is clean, rebooted and did 
`sudo systemctl restart docker`

Then cuda errors

```
sudo docker exec -it 7abde00f1842 /bin/bash

________                               _______________                
___  __/__________________________________  ____/__  /________      __
__  /  _  _ \_  __ \_  ___/  __ \_  ___/_  /_   __  /_  __ \_ | /| / /
_  /   /  __/  / / /(__  )/ /_/ /  /   _  __/   _  / / /_/ /_ |/ |/ / 
/_/    \___//_/ /_//____/ \____//_/    /_/      /_/  \____/____/|__/


You are running this container as user with ID 1000 and group 1000,
which should map to the ID and group for your user on the Docker host. Great!

tf-docker /tf > cd /home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src/
tf-docker /home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src > python main.py --source=bentham --train
2019-12-26 23:01:13.422419: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.6
2019-12-26 23:01:13.446401: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer_plugin.so.6
2019-12-26 23:01:19.131157: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2019-12-26 23:01:19.163713: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.164228: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.755GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2019-12-26 23:01:19.164285: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 23:01:19.164339: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 23:01:19.181939: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2019-12-26 23:01:19.186051: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2019-12-26 23:01:19.219655: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2019-12-26 23:01:19.222981: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2019-12-26 23:01:19.223009: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 23:01:19.223076: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.223463: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.223765: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2019-12-26 23:01:19.224099: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-12-26 23:01:19.250610: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3696000000 Hz
2019-12-26 23:01:19.251627: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x8035b20 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2019-12-26 23:01:19.251648: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2019-12-26 23:01:19.344922: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.345338: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7e7bd20 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2019-12-26 23:01:19.345355: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): GeForce RTX 2080 Ti, Compute Capability 7.5
2019-12-26 23:01:19.345489: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.345810: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce RTX 2080 Ti computeCapability: 7.5
coreClock: 1.755GHz coreCount: 68 deviceMemorySize: 10.76GiB deviceMemoryBandwidth: 573.69GiB/s
2019-12-26 23:01:19.345842: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 23:01:19.345857: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 23:01:19.345873: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2019-12-26 23:01:19.345888: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2019-12-26 23:01:19.345903: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2019-12-26 23:01:19.345917: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2019-12-26 23:01:19.345931: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 23:01:19.345977: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.346316: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.346639: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2019-12-26 23:01:19.347084: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
2019-12-26 23:01:19.924373: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1096] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-12-26 23:01:19.924403: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102]      0 
2019-12-26 23:01:19.924410: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] 0:   N 
2019-12-26 23:01:19.924955: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.925353: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-26 23:01:19.925701: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1241] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 9581 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5)
Model: "model"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input (InputLayer)           [(None, 1024, 128, 1)]    0         
_________________________________________________________________
conv2d (Conv2D)              (None, 512, 64, 16)       160       
_________________________________________________________________
p_re_lu (PReLU)              (None, 512, 64, 16)       16        
_________________________________________________________________
batch_normalization (BatchNo (None, 512, 64, 16)       112       
_________________________________________________________________
full_gated_conv2d (FullGated (None, 512, 64, 16)       4640      
_________________________________________________________________
conv2d_1 (Conv2D)            (None, 512, 64, 32)       4640      
_________________________________________________________________
p_re_lu_1 (PReLU)            (None, 512, 64, 32)       32        
_________________________________________________________________
batch_normalization_1 (Batch (None, 512, 64, 32)       224       
_________________________________________________________________
full_gated_conv2d_1 (FullGat (None, 512, 64, 32)       18496     
_________________________________________________________________
conv2d_2 (Conv2D)            (None, 256, 16, 40)       10280     
_________________________________________________________________
p_re_lu_2 (PReLU)            (None, 256, 16, 40)       40        
_________________________________________________________________
batch_normalization_2 (Batch (None, 256, 16, 40)       280       
_________________________________________________________________
full_gated_conv2d_2 (FullGat (None, 256, 16, 40)       28880     
_________________________________________________________________
dropout (Dropout)            (None, 256, 16, 40)       0         
_________________________________________________________________
conv2d_3 (Conv2D)            (None, 256, 16, 48)       17328     
_________________________________________________________________
p_re_lu_3 (PReLU)            (None, 256, 16, 48)       48        
_________________________________________________________________
batch_normalization_3 (Batch (None, 256, 16, 48)       336       
_________________________________________________________________
full_gated_conv2d_3 (FullGat (None, 256, 16, 48)       41568     
_________________________________________________________________
dropout_1 (Dropout)          (None, 256, 16, 48)       0         
_________________________________________________________________
conv2d_4 (Conv2D)            (None, 128, 4, 56)        21560     
_________________________________________________________________
p_re_lu_4 (PReLU)            (None, 128, 4, 56)        56        
_________________________________________________________________
batch_normalization_4 (Batch (None, 128, 4, 56)        392       
_________________________________________________________________
full_gated_conv2d_4 (FullGat (None, 128, 4, 56)        56560     
_________________________________________________________________
dropout_2 (Dropout)          (None, 128, 4, 56)        0         
_________________________________________________________________
conv2d_5 (Conv2D)            (None, 128, 4, 64)        32320     
_________________________________________________________________
p_re_lu_5 (PReLU)            (None, 128, 4, 64)        64        
_________________________________________________________________
batch_normalization_5 (Batch (None, 128, 4, 64)        448       
_________________________________________________________________
max_pooling2d (MaxPooling2D) (None, 128, 2, 64)        0         
_________________________________________________________________
reshape (Reshape)            (None, 128, 128)          0         
_________________________________________________________________
bidirectional (Bidirectional (None, 128, 256)          198144    
_________________________________________________________________
time_distributed (TimeDistri (None, 128, 128)          32896     
_________________________________________________________________
bidirectional_1 (Bidirection (None, 128, 256)          198144    
_________________________________________________________________
time_distributed_1 (TimeDist (None, 128, 98)           25186     
=================================================================
Total params: 692,850
Trainable params: 691,570
Non-trainable params: 1,280
_________________________________________________________________
Train for 545 steps, validate for 85 steps
Epoch 1/1000
2019-12-26 23:01:27.960566: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2019-12-26 23:01:28.353452: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2019-12-26 23:01:29.606042: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2019-12-26 23:01:29.617482: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2019-12-26 23:01:29.617918: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Unknown: Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[{{node model/conv2d/Conv2D}}]]
	 [[loss/time_distributed_1_loss/scan/while/body/_1/Less/_64]]
2019-12-26 23:01:29.618012: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Unknown: Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[{{node model/conv2d/Conv2D}}]]
  1/545 [..............................] - ETA: 59:56WARNING:tensorflow:Can save best model only with val_loss available, skipping.
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: 
WARNING:tensorflow:Reduce LR on plateau conditioned on metric `val_loss` which is not available. Available metrics are: lr
  1/545 [..............................] - ETA: 1:00:30Traceback (most recent call last):
  File "main.py", line 155, in <module>
    verbose=1)
  File "/home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src/network/model.py", line 175, in fit
    use_multiprocessing=use_multiprocessing, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training.py", line 819, in fit
    use_multiprocessing=use_multiprocessing)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 342, in fit
    total_epochs=epochs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 128, in run_one_epoch
    batch_outs = execution_function(iterator)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 98, in execution_function
    distributed_function(input_fn))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 568, in __call__
    result = self._call(*args, **kwds)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 632, in _call
    return self._stateless_fn(*args, **kwds)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 2363, in __call__
    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 1611, in _filtered_call
    self.captured_inputs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 1692, in _call_flat
    ctx, args, cancellation_manager=cancellation_manager))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 545, in call
    ctx=ctx)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/execute.py", line 67, in quick_execute
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.UnknownError: 2 root error(s) found.
  (0) Unknown:  Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[node model/conv2d/Conv2D (defined at /home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src/network/model.py:175) ]]
	 [[loss/time_distributed_1_loss/scan/while/body/_1/Less/_64]]
  (1) Unknown:  Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[node model/conv2d/Conv2D (defined at /home/john/Documents/GitHub/Offline_Handwriting_Recognition/Solutions/arthurflor23_handwritten-text-recognition/handwritten-text-recognition-master/src/network/model.py:175) ]]
0 successful operations.
0 derived errors ignored. [Op:__inference_distributed_function_17754]

Function call stack:
distributed_function -> distributed_function

2019-12-26 23:01:29.771757: W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Cancelled: Operation was cancelled

```

**Comment by johngrabner** [comment]
### Still crashes

```
2019-12-27 00:08:54.931864: E tensorflow/stream_executor/cuda/cuda_dnn.cc:329] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2019-12-27 00:08:54.932313: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Unknown: Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above.
	 [[{{node model/conv2d/Conv2D}}]]
	 [[loss/time_distributed_1_loss/GatherNd/_102]]
```

after adding the following to my dockerfile (ie to match the fixes you shared)

```
FROM tensorflow/tensorflow:2.1.0rc2-gpu-py3-jupyter

# to see if  #35434 fixes the issues
ARG CUDA=10.1
RUN apt-get update && apt-get install -y --no-install-recommends \
        cuda-nvrtc-${CUDA/./-} \
        cuda-nvrtc-dev-${CUDA/./-} \
        libcublas10=10.2.1.243-1 \ 
        libcublas-dev=10.2.1.243-1 
```

### Maybe this code is stressing  cuda differently

I tried a simple 10 line nmist example and it works correctly with gpu, even before the last change and with only the first change.

The code that is causing the above error is sourced from https://github.com/arthurflor23/handwritten-text-recognition.  I am not too familiar with it. Author claims its solid.

Arthur runs this code on colab with 2.1.0-rc1.  I am running this on docker and 
it only worked for me with a cpu docker upgraded to 2.1.0.rc1 and not with a gpu docker.

As you can see it fails with the GPU version even with these fixes.  

I use the GPU docker all of the time, but with 1.13.1-gpu-py3-jupyter on other code.
Arthur's code that is dying, uses the new 2.0 api.

### More Robust Test:

Do you have a more robust test case I can run, my setup is not too special, but 
maybe you don't have this at your location.
- GeForce RTX 2080 Ti/PCIe/SSE2
- Intel® Core™ i7-8700K CPU @ 3.70GHz
- Ubuntu 18.04.3 LTS

Any how, let me know how I can help.

**Comment by johngrabner** [comment]
still broken

**Comment by angerson** [comment]
The original issue of missing docker images is now resolved. For other build problems, please file a new issue.

**Comment by johngrabner** [comment]
A workaround for second issue was made available by https://github.com/tensorflow/tensorflow/issues/25446.

Looks like tf 2.0 stresses cuda more.

https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth

So any psudo complex tf using rtx video cards will need this as the first few lines of code.

```
gpus = tf.config.experimental.list_physical_devices('GPU')
if gpus:
  try:
    # Currently, memory growth needs to be the same across GPUs
    for gpu in gpus:
      tf.config.experimental.set_memory_growth(gpu, True)
    logical_gpus = tf.config.experimental.list_logical_devices('GPU')
    print(len(gpus), "Physical GPUs,", len(logical_gpus), "Logical GPUs")
  except RuntimeError as e:
    # Memory growth must be set before GPUs have been initialized
    print(e)
```
